zerver/decorator.py

```diff
@@ -597,7 +597,7 @@ def wrapped_func(request, *args, **kwargs):
 
             try:
                 user = request.user
-            except:
+            except AttributeError:
```

Comment:
This should probably also be an `except Exception`.  The reason is subtle -- while it might appear as though the only failure possible here is an AttributeError, `request.user` is a property that calls variable functions, and thus may be able to throw other exceptions.  And the handler code is clearly intended to catch any issue here so that the following lines can run properly.